destination/src/main/scala/quasar/plugin/sqlserver/destination/CsvUpsertSink.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.plugin.sqlserver.destination
+
+import slamdata.Predef._
+
+import quasar.plugin.sqlserver._
+
+import quasar.api.push.OffsetKey
+import quasar.connector.destination.ResultSink.UpsertSink
+import quasar.connector.render.RenderConfig
+import quasar.connector.{DataEvent, IdBatch, MonadResourceErr}
+import quasar.lib.jdbc._
+import quasar.lib.jdbc.destination.{WriteMode => JWriteMode}
+
+import java.lang.CharSequence
+
+import cats.effect.Effect
+import cats.data.NonEmptyVector
+import cats.implicits._
+
+import doobie._
+import doobie.free.connection.{commit, rollback, setAutoCommit, unit}
+import doobie.implicits._
+import doobie.util.transactor.Strategy
+
+import fs2.Pipe
+import org.slf4s.Logger
+
+import skolems.∀
+
+private[destination] object CsvUpsertSink {
+  type Consume[F[_], A] =
+    Pipe[F, DataEvent[CharSequence, OffsetKey.Actual[A]], OffsetKey.Actual[A]]
+
+  def builder[F[_]: Effect: MonadResourceErr](
```

Comment:
Other than the difference in config and `Delete` event handling, this looks identical to the append sink, could we unify them as this impl would be sufficient for appends, it'd just never see a `Delete` in that case.

Reply:
:+1: